src/Microsoft.Azure.SignalR.Common/ServiceCredential.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.SignalR.Common
+{
+    public class ServiceCredential
+    {
+        public string Endpoint { get; private set; }
+        public string AccessKey { get; private set; }
+        public string Version { get; private set; }
+        public int? Port { get; private set; }
+
+        public ServiceCredential()
+        {
+        }
+
+        public ServiceCredential(string connectionString)
+        {
+            Parse(connectionString);
+        }
+
+        public void Parse(string connectionString)
```

Comment:
Better to be a static method that returns parsed object?

Reply:
Return the new class now